### ISSUE TITLE
Example shader/array_texture crash with wgpu error

### ISSUE THREAD

**Issue by giusdp** [bevyengine/bevy#2122]
## Bevy version

v0.5.0 commit 7d0e98f34c43eb62522c0c65de64f7b458b899be

## Operating system & version

Manjaro Linux

## What you did

Ran the example array_texture with `cargo run --example array_texture` 

## What you expected to happen

The example to work.

## What actually happened

After opening the window at first, it quickly crashed.

## Additional information
The console log errors:

```
wgpu error: Validation Error

Caused by:
    In Device::create_bind_group
    texture binding 0 expects dimension = D2, but given a view with dimension = D2Array


thread 'Compute Task Pool (0)' panicked at 'Handling wgpu errors as fatal by default', /home/giusdp/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.8.0/src/backend/direct.rs:1955:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'task has failed', /home/giusdp/.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.0.3/src/task.rs:368:45
zsh: segmentation fault (core dumped)  cargo run --example array_texture
```

**Comment by MinerSebas** [comment]
Bisecting the Repo, that Panic appeared after the move to wgpu 0.8 in #1959

**Comment by mockersf** [comment]
@jakobhellermann I think it's an issue in reflexion, would #2130 also fix this case?

**Comment by jakobhellermann** [comment]
> @jakobhellermann I think it's an issue in reflection, would #2130 also fix this case?

Just tried it, with #2130 the example works again.

**Comment by superdump** [comment]
#2130 fixes this for me too on macOS.

**Comment by KoomZog** [comment]
I'm having the same issue as OP. I'm on Bevy main (on win 10). This is the error message I got: https://pastebin.com/iVbceafu
Is there a simple fix for this, or should I just wait for 0.6?

**Comment by cart** [comment]
Closing this because the old renderer has been removed and the new renderer relies way less on reflection, so this shouldn't be an issue anymore. Feel free to open a new issue if you still have problems.
